secret-integers/src/lib.rs

```diff
@@ -146,19 +146,19 @@ macro_rules! define_unary_op {
 }
 
 macro_rules! define_shift {
-    ($name:ident, $op:tt, $op_name:ident, $func_op:ident, $assign_name:ident, $assign_func:ident) => {
-        impl $op_name<u32> for $name {
+    ($name:ident, $op:tt, $wrapop:ident, $op_name:ident, $func_op:ident, $assign_name:ident, $assign_func:ident) => {
+        impl $op_name<usize> for $name {
             type Output = Self;
             #[inline]
-            fn $func_op(self, rhs: u32) -> Self {
+            fn $func_op(self, rhs: usize) -> Self {
                 let $name(i1) = self;
-                $name(i1 $op rhs)
```

Comment:
So here we're changing the default behaviour of secret integers shifting to be `wrapping` ? What does it even mean for a shift to be wrapping ? Does it mean rotating instead of shifting ?

Reply:
Right this changes the behavior. But no, it's not rotating. It's only not panicking in case bits are getting shifted out of the value (which is really what we want). Also note that the panic is only in debug builds. For optimized builds the behavior should actually be the same.

Reply:
Ah yes indeed, we don't want the shifts to panick when the bits are shifted out.